apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -420,3 +422,17 @@ func (r *OpenStackControlPlane) DefaultServices() {
 	// Barbican
 	r.Spec.Barbican.Template.Default()
 }
+
+// DefaultLabel - adding default label to the OpenStackControlPlane
+func (r *OpenStackControlPlane) DefaultLabel() {
+	// adds map[string]string{"core.openstack.org/openstackcontrolplane": r.name>} to the
+	// instance, if not already provided in the CR. With this ctlplane object can be
+	// queried using the default label.
+	typeLabel := strings.ToLower(r.GroupVersionKind().Group + "/" + r.Kind)
+	if _, ok := r.Labels[typeLabel]; !ok {
+		if r.Labels == nil {
+			r.Labels = map[string]string{}
+		}
+		r.Labels[typeLabel] = r.Name
```

Comment:
maybe don't have to set `r.Name`. when you know the name you could query it direct